controllers/helmchart_controller.go

```diff
@@ -426,16 +435,147 @@ func (r *HelmChartReconciler) reconcileFromTarballArtifact(ctx context.Context,
 		return sourcev1.HelmChartNotReady(chart, sourcev1.StorageOperationFailedReason, err.Error()), err
 	}
 
-	// Read the chart metadata
-	chartMetadata, err := chartutil.LoadChartfile(path.Join(chartPath, chartutil.ChartfileName))
+	// Read the chart
+	helmChart, err := loader.LoadDir(chartPath)
+	if err != nil {
+		err = fmt.Errorf("load chart error: %w", err)
+		return sourcev1.HelmChartNotReady(chart, sourcev1.StorageOperationFailedReason, err.Error()), err
+	}
+
+	// Load dependencies
+	deps := helmChart.Metadata.Dependencies
+	lock := helmChart.Lock
+	if lock != nil {
+		// Load from lockfile if exists
+		deps = lock.Dependencies
+	}
+
+	// Create charts directory inside chart path
+	depChartsPath := path.Join(chartPath, "charts")
+	err = os.MkdirAll(depChartsPath, 0755)
 	if err != nil {
-		err = fmt.Errorf("load chart metadata error: %w", err)
+		err = fmt.Errorf("unable to create dependencies charts directory: %w", err)
 		return sourcev1.HelmChartNotReady(chart, sourcev1.StorageOperationFailedReason, err.Error()), err
 	}
 
+	for _, dep := range deps {
+		if strings.HasPrefix(dep.Repository, "file://") {
```

Comment:
Most of this was borrowed from: https://github.com/qwangqw/helm/blob/master/pkg/downloader/manager.go#L533